hw/xbox/nv2a/pgraph.c

```diff
@@ -5639,6 +5664,9 @@ static void pgraph_upload_surface_data(NV2AState *d, SurfaceBinding *surface,
     glBindTexture(GL_TEXTURE_2D, last_texture_binding);
 
     pgraph_bind_current_surface(d);
+    /*the surfaces must be always updated in the structure for the texture exporter.
+    Hence why the export is called on each upload when is done*/
+   export_pgraph_data(d, surface);
```

Comment:
I assume you're still working out how to make this only dump when you want it; calling this on every upload will have a very significant performance impact.